email_backend/src/email_backend.js

```diff
@@ -143,6 +147,40 @@ const make_email_backend = (templates) => {
         );
         if (mail_sent_successfully){
           response.send("200");
+
+          //Sending info to MongoDB server
+          const mongoose = require('mongoose');
+          const Schema = mongoose.Schema;
+          
+          mongoose.connect('mongodb://localhost/Feedback');
+
+          mongoose.connection.once('open', function(){
```

Comment:
If you haven't seen promises much yet, let me know. They're another point I can give a bit of history on.

Reply:
I've read a bit up on promises too, but also not too knowledgable on the history of its use aha 
I'll change it to your way for now though

Reply:
Long story. Didn't read all of it, but googled around and I think this article breaks down callbacks vs promises vs async/await well: https://blog.avenuecode.com/callback-hell-promises-and-async/await

Read that, or at least skim it, before this next bit, because I'm not going to redefine everything here.

Short of the history is that asynchronus JavaScript used to be totally callback-based and it was a nightmare to read. Good luck maintaining someone else's callback based code, let alone your own. The phrase "callback hell" became well known.

In 2015 (big year for JS, we got ECMAScript 6 which added a LOT of features) we got promises. They've got a bit more going on, probably harder to learn up-front but that's just because they formalize a lot of what devs were doing with adhoc, hacky, callback techniques. The reason history is important here, IMO, is because it's hard to buy that the apparently more complex promise approach is better than callbacks if you haven't experienced callback hell your self.

Async/await is even newer than promises (2017). Nicer syntax, your code pretty much looks like standard, totally synchronous JS with a couple async/await keywords thrown in. Under the hood, it's all promises still, and the boundary between async functions and synchronus JS execution still needs to be a promise or callback, so you've got to be comfortable with promises before you use async/await IMO.